test/interpolating_tests.jl

```diff
@@ -100,6 +104,11 @@ function adjoint_linear(u, p, t, sol)
     return -[0 b; -a 0] * u - 2.0 * (sol(t) .- 1.0)
 end
 
+function adjoint_linear_inplace(du, u, p, t, sol)
+    a, b = p
+    du .= -[0 b; -a 0] * u - 2.0 * (sol(t) .- 1.0)
```

Comment:
This also has the allocation right before the `.=` so it's not saving an allocation